lib/acme/client/resources/authorization.rb

```diff
@@ -7,12 +7,26 @@ class Acme::Client::Resources::Authorization
 
   def initialize(client, response)
     @client = client
-    assign_challenges(response.body['challenges'])
+    @uri = response.headers['Location']
     assign_attributes(response.body)
   end
 
+  def verify_status
+    response = @client.connection.get(@uri)
+
+    assign_attributes(response.body)
+    status
+  end
```

Comment:
I assume some users will need to be able to serialize / load this object now that you can query on it?